### ISSUE TITLE
Change Access Denied page for Galaxy Workflows

### ISSUE THREAD

**Issue by mestato** [statonlab/hardwoods_site#280]
Add a note that "Only registered users are allowed to run Galaxy workflows. Please [log in] or [register an account]"

**Comment by almasaeed2010** [comment]
This might be harder than it sounds. One solution I can think of is to change all 403 pages to have a login form/link if the user is not logged in (which can be good for many other reasons). We can do that on a theme level but that means not having this feature in the module itself.

@MingChen0919 how is the page implemented? Is it a basic page on the site built with the GUI? Or is it programmatically generated?

Thanks

**Comment by MingChen0919** [comment]
I didn't implement these pages. I think they are regular 403 pages.

**Comment by MingChen0919** [comment]
if you change all 403 pages, would it be that all 403 pages will display this message? but we only want to show the message on galaxy workflows. Would adding a block be a possible solution?

**Comment by almasaeed2010** [comment]
We wouldn't add this particular message. It'll be more generic:

> Access Denied
> You are currently not logged in. You may need to [log in](#login) to view this page

**Comment by bradfordcondon** [comment]
this would only be bad if the 403 error was for an admin resource.

If the 403 error is for any other resource, then directing them to make an account is the desired behavior.

**Comment by almasaeed2010** [comment]
Even the user is logged in and they still don't have access (such as trying to get into an admin page):
> Access Denied
> You do not have permissions to access this page

**Comment by bradfordcondon** [comment]
yeah its fine i guess.  i was thinking its not great to prompt a user to log in, and then, when they do, tell them they dont have access.  However I've seen that exact behavior on other sites.

**Comment by MingChen0919** [comment]
I see. thanks for the explanation!
